Src/FluentAssertions/Equivalency/EquivalencyValidator.cs

```diff
@@ -31,38 +31,38 @@ public void RecursivelyAssertEquality(Comparands comparands, IEquivalencyValidat
     {
         var scope = AssertionScope.Current;
 
-        if (ShouldCompareNodesThisDeep(context.CurrentNode, context.Options, scope))
+        if (ShouldContinueThisDeep(context.CurrentNode, context.Options, scope))
         {
-            UpdateScopeWithReportableContext(scope, comparands, context.CurrentNode);
+            TrackWhatWeNeedToProvideContextToFailures(scope, comparands, context.CurrentNode);
 
             if (!context.IsCyclicReference(comparands.Expectation))
             {
-                RunStepsUntilEquivalencyIsProven(comparands, context);
+                TryToProveNodesAreEquivalent(comparands, context);
             }
         }
     }
 
-    private static bool ShouldCompareNodesThisDeep(INode currentNode, IEquivalencyAssertionOptions options,
+    private static bool ShouldContinueThisDeep(INode currentNode, IEquivalencyAssertionOptions options,
         AssertionScope assertionScope)
     {
         bool shouldRecurse = options.AllowInfiniteRecursion || currentNode.Depth <= MaxDepth;
-
         if (!shouldRecurse)
         {
+            // This will throw, unless we're in using (new AssertionScope())
```

Comment:
Not sure the if the "using" here is meant as the `using` keyword or the regular English word.

Alternatives
"This will throw, unless we're using an AssertionScope"
"This will throw, unless we're inside an AssertionScope"